### ISSUE TITLE
Possible bug/omission: APIRequestException not raised on status code 420

### ISSUE THREAD

**Issue by thecnoNSMB** [blha303/srcomapi#14]
Speedrun.com uses status code 420 to indicate that the client is being rate-limited. Details [here](https://github.com/speedruncomorg/api/blob/master/throttling.md).

**Comment by thecnoNSMB** [comment]
I should've suspected something when I typed "I think it's the library's fault that my code doesn't work." It's my fault and I'm an idiot. Still, I don't see any handling for this in the source, and this code should probably be being handled if it isn't already.

**Comment by blha303** [comment]
At the time I wrote srcomapi that throttling didn't exist. I'll add it to the list of things to improve.

**Comment by thecnoNSMB** [comment]
Apologies for being an idiot, in any case.

**Comment by blha303** [comment]
No, you're fine, I should be catching throttling. I'll figure something out to fix it.
